osgar/tools/lidarview.py

```diff
@@ -432,6 +434,20 @@ def lidarview(gen, caption_filename, callback=False, out_video=None, jump=None):
 
             foreground.fill((0, 0, 0))
             img = image2 if use_image2 else image
+            if callback_img:
+                if last_timestamp != timestamp:
+                    # convert pygame image to numpy image
+                    view = pygame.surfarray.array3d(img)
+                    view = view.transpose([1, 0, 2])
+                    np_img = cv2.cvtColor(view, cv2.COLOR_RGB2BGR)
+                    np_img = callback_img.run_on_image(np_img)
```

Comment:
Oh, I see now - well then you can pass really only the callback function which should be `callback_img.run_on_image`.